build.py

```diff
@@ -94,6 +92,48 @@ def __init__(self, name, params, dut):
         self.top = args.top
         # the tests parameters
         self.params = params # FIXME ABD
+
+
+        # Load configuration from JSON file or use defaults
+        self.load_config()
+
+    def load_config(self):
+        # Default JSON file location
+        json_directory = 'app/cfg/'
+
+        # Check if the -cfg flag is provided
+        if args.cfg:
+            json_file = os.path.join(json_directory, args.cfg +'.json')
+            if not os.path.exists(json_file):
+                print_message(f'[ERROR] There is no file \'{args.cfg}\'')
+                exit(1)
+            else:
+                print_message(f'[INFO] Using configuration from \'{args.cfg}\'')
+                 #loading configuration from the specified JSON file
+                with open(json_file) as config_file:
+                    config_data = json.load(config_file)
+                # Update configuration as defined in the JSON file
+                Test.I_MEM_OFFSET = str(config_data['I_MEM_OFFSET'])
```

Comment:
This is fine but from a coding style point of view, it feels like there is a way to not duplicate the:
```
            Test.I_MEM_OFFSET = str(config_data['I_MEM_OFFSET'])
            Test.I_MEM_LENGTH = str(config_data['I_MEM_LENGTH'])
            Test.D_MEM_OFFSET = str(config_data['D_MEM_OFFSET'])
            Test.D_MEM_LENGTH = str(config_data['D_MEM_LENGTH'])
            Test.crt0_file    = config_data['crt0_file'] 
            Test.rv32_gcc     = config_data['rv32_gcc']
            Test.name         = config_data['name']    
```

Try to see if you have an idea how it can done. (Sure Chat GPT can suggest some things)